angular/src/lib/sidebar/sidebar.component.ts

```diff
@@ -1,23 +1,42 @@
-import { Component, OnInit, Input, ElementRef } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output, OnDestroy } from '@angular/core';
+import { fromEvent, Subscription } from 'rxjs';
+import { debounceTime, map } from 'rxjs/operators';
 import { SidebarService } from './sidebar.service';
 
 @Component({
   selector: 'md-sidebar',
   template: `
 
-  <div class="md-sidebar" [ngClass]="sidebarClass">
+  <div class="md-sidebar" [ngClass]="[
+    theme ? 'md-sidebar--' + theme : '',
+    isFixed ? 'md-sidebar--fixed' : '',
+    !isPageLevel ? 'md-sidebar--global' : '',
+    withTopbar ? 'md-sidebar--topbar' : '',
+    withIcons && !isPageLevel ? 'md-sidebar--indented' : '',
+    hasTier ? 'md-sidebar--nested' : '',
+    expanded ? 'md-sidebar--expanded' : '',
+    !expanded && withIcons ? 'md-sidebar--minimized' : '',
+    !expanded && !withIcons ? 'md-sidebar--collapsed' : '',
+    wrapperClass
+  ]">
```

Comment:
Good to know this improves performance from change detection, will have to update existing component getters.